Apps/Contoso.iOS.Puppet/AppDelegate.cs

```diff
@@ -1,86 +1,95 @@
-﻿using System;
+﻿#pragma warning disable XI0002 // Notifies you from using newer Apple APIs when targeting an older OS version
+
+using System;
+using System.Linq;
 using Foundation;
 using Microsoft.AppCenter;
 using Microsoft.AppCenter.Analytics;
 using Microsoft.AppCenter.Crashes;
 using Microsoft.AppCenter.Distribute;
 using Microsoft.AppCenter.Push;
 using UIKit;
+using UserNotifications;
 
 namespace Contoso.iOS.Puppet
 {
-    // The UIApplicationDelegate for the application. This class is responsible for launching the
-    // User Interface of the application, as well as listening (and optionally responding) to application events from iOS.
     [Register("AppDelegate")]
     public class AppDelegate : UIApplicationDelegate
     {
-        // class-level declarations
-
         public override UIWindow Window
         {
             get;
             set;
         }
 
+        static bool _didTapNotification;
+
         public override bool FinishedLaunching(UIApplication application, NSDictionary launchOptions)
         {
-            // Override point for customization after application launch.
-            // If not required for your application you can safely delete this method
-
             AppCenter.LogLevel = LogLevel.Verbose;
             AppCenter.SetLogUrl("https://in-integration.dev.avalanch.es");
+            if (UIDevice.CurrentDevice.CheckSystemVersion(10, 0))
+            {
+                UNUserNotificationCenter.Current.Delegate = new PushDelegate();
+            }
+            Push.PushNotificationReceived += (sender, e) =>
+            {
+                string message = "";
+                if (UIDevice.CurrentDevice.CheckSystemVersion(10, 0))
+                {
+                    if (_didTapNotification)
```

Comment:
The ternary operator can improve readability here.
`message += _didTapNotification ? "Tapped notification\n" : "Received in foreground\n";`
This also removes `message.Length > 0` below

Reply:
Will use ternary but on iOS 9 the second suggestion will add an extra new line so it's slightly different.

Reply:
If message is null but have custom data we end up having 1 extra newline, this needs a bit more checking